test/ui-test/src/screens/StatusMainScreen.py

```diff
@@ -44,10 +44,18 @@ def __init__(self):
         verify_screen(MainScreenComponents.PUBLIC_CHAT_ICON.value)
 
     # Wait for the banner to disappear otherwise the click might land badly
-    def wait_for_banner_to_disappear():
-        [bannerLoaded, _] = is_loaded_visible_and_enabled(MainScreenComponents.MODULE_WARNING_BANNER.value)
-        if (bannerLoaded):
-            time.sleep(5)
+    def wait_for_banner_to_disappear(timeoutMSec: int = 3000):
+        start = time.time()
+        while(start + timeoutMSec / 1000 > time.time()):
+            try:
+                obj = wait_and_get_obj(MainScreenComponents.MODULE_WARNING_BANNER.value, 100)
+                if not obj.visible:
+                    return
+                obj.close()
+            except:
+                return
+            squish.snooze(0.1)
+        raise Exception(f"Banner is still visible after {timeoutMSec}ms")
```

Comment:
The is no validation needed because we know that this is a failure. Therefore I used the universal failing mechanics I see squish and python use. And it is shown nicely in the tests. However, I now added `fail_test` to SquishDriver and use that instead, which calls squish test's`fail`. Not sure if this is a better way but let's see how it works in practice.